libs/messenger/src/messenger/mailbox.cpp

```
@@ -0,0 +1,171 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "messenger/messenger_api.hpp"
+#include "messenger/messenger_protocol.hpp"
+
+#include <cassert>
+
+namespace fetch {
+namespace messenger {
+
+Mailbox::Mailbox(muddle::MuddlePtr &muddle)
+  : message_endpoint_{muddle->GetEndpoint()}
+  , message_subscription_{
+        message_endpoint_.Subscribe(SERVICE_MSG_TRANSPORT, CHANNEL_MESSEGNER_TRANSPORT)}
+{
+  message_subscription_->SetMessageHandler(this, &Mailbox::OnNewMessagePacket);
+}
+
+void Mailbox::OnNewMessagePacket(muddle::Packet const &packet, Address const & /*last_hop*/)
+{
+  fetch::serializers::MsgPackSerializer serialiser(packet.GetPayload());
+  try
+  {
+    Message message;
+    serialiser >> message;
+    SendMessage(message);
+  }
+  catch (...)
+  {
```

Empty catch bad practice

---

:+1: